go/cmd/dolt/commands/remote.go

```diff
@@ -207,21 +225,97 @@ func parseRemoteArgs(apr *argparser.ArgParseResults, scheme, remoteUrl string) (
 	return params, nil
 }
 
-func printRemotes(dEnv *env.DoltEnv, apr *argparser.ArgParseResults) errhand.VerboseError {
-	remotes, err := dEnv.GetRemotes()
+func callSQLAdd(sqlCtx *sql.Context, queryist cli.Queryist, remoteName, remoteUrl string) error {
+	qry, err := dbr.InterpolateForDialect("call dolt_remote('add', ?, ?)", []interface{}{remoteName, remoteUrl}, dialect.MySQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = GetRowsForSql(queryist, sqlCtx, qry)
+	return err
+}
 
+// callSQLRemove calls the SQL function `call `dolt_remote('remove', remoteName)`
+func callSQLRemove(sqlCtxe *sql.Context, queryist cli.Queryist, remoteName string) error {
+	qry, err := dbr.InterpolateForDialect("call dolt_remote('remove', ?)", []interface{}{remoteName}, dialect.MySQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = GetRowsForSql(queryist, sqlCtxe, qry)
+	return err
+}
+
+type remote struct {
+	Name   string
+	Url    string
+	Params string
+}
+
+func getRemotesSQL(sqlCtx *sql.Context, queryist cli.Queryist) ([]remote, error) {
+	qry := "select name,url,params from dolt_remotes"
+	rows, err := GetRowsForSql(queryist, sqlCtx, qry)
+	if err != nil {
+		return nil, err
+	}
+
+	remotes := make([]remote, 0, len(rows))
+	for _, r := range rows {
+		name, ok := r[0].(string)
+		if !ok {
+			return nil, fmt.Errorf("invalid remote name")
+		}
+
+		url, ok := r[1].(string)
+		if !ok {
+			return nil, fmt.Errorf("invalid remote url")
+		}
+
+		params, err := getJsonAsString(sqlCtx, r[2])
+		if err != nil {
+			return nil, fmt.Errorf("invalid params")
+		}
+
+		remotes = append(remotes, remote{
+			Name:   name,
+			Url:    url,
+			Params: params,
+		})
+	}
+
+	return remotes, nil
+}
+
+// getJsonAsString returns a string representation of the given interface{}, which can either be a string or a JSONDocument.
+// If it is a string, it gets returned as is. If it is a JSONDocument, it gets converted to a string.
+// SQLEngine and the remote connection behave a little differently here, which is the reason for needing this.
+func getJsonAsString(sqlCtx *sql.Context, params interface{}) (string, error) {
+	switch p := params.(type) {
+	case string:
+		return p, nil
+	case types.JSONDocument:
```

Comment:
It's better to check if it's a `types.SQLWrapper` here and then pass it to `types.StringifyJSON` to get the string. This guarantees it will work if a later change causes the query to  return a different `SQLWrapper` implementation.